checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -133,166 +411,116 @@ protected boolean checkOverride(MethodTree overriderTree,
-    protected void checkPreconditions(Tree tree,
-            Element invokedElement, boolean methodCall, Set<Pair<String, String>> additionalPreconditions) {
+        if (treeKind == Kind.MEMBER_SELECT) {
+            Element treeElement = TreeUtils.elementFromUse(tree);
 
-        if (additionalPreconditions == null) {
-            additionalPreconditions = new HashSet<>();
+            if (treeElement != null && treeElement.getKind() == ElementKind.METHOD) { // Method calls are not dereferences.
+                isMethodCall = true;
+            }
         }
 
-        // Retrieve the @GuardedBy annotation on the receiver of the enclosing method
-        // and add it to the preconditions set if the receiver on the enclosing method is
-        // the same as the receiver on the field/method we are checking preconditions for.
-        MethodTree enclosingMethod = TreeUtils.enclosingMethod(atypeFactory.getPath(tree));
-
-        if (enclosingMethod != null) {
-            ExecutableElement methodElement = TreeUtils.elementFromDeclaration(enclosingMethod);
-
-            if (methodElement != null) {
-                TypeMirror rt = methodElement.getReceiverType();
-
-                if (rt != null) {
-                    List<? extends AnnotationMirror> amList = rt.getAnnotationMirrors();
-
-                    for (AnnotationMirror annotationMirror : amList) {
-
-                        if (AnnotationUtils.areSameByClass( annotationMirror, checkerGuardedByClass) ||
-                            AnnotationUtils.areSameByClass( annotationMirror, javaxGuardedByClass) ||
-                            AnnotationUtils.areSameByClass( annotationMirror, jcipGuardedByClass)) {
-                            List<String> guardedByValue = AnnotationUtils.getElementValueArray(annotationMirror, "value", String.class, false);
+        if (!isMethodCall) {
+            ExpressionTree expr = null;
+
+            switch(treeKind) {
+                case ARRAY_ACCESS:
+                    expr = ((ArrayAccessTree) tree).getExpression();
+                    break;
+                case MEMBER_SELECT:
+                    expr = ((MemberSelectTree) tree).getExpression();
+                    break;
+                default:
+                    expr = tree;
+                    break;
+            }
 
-                            // A GuardedBy annotation on the receiver of the enclosing method was found.
-                            // Now check if the receiver on the enclosing method is the same as the receiver
-                            // on the field/method we are checking preconditions for.
+            Element invokedElement = TreeUtils.elementFromUse(expr);
 
-                            if (guardedByValue != null) {
-                                Node nodeNode = atypeFactory.getNodeForTree(tree);
-                                Node receiverNode = null;
+            AnnotatedTypeMirror atmOfReceiverOrIdentifier = atypeFactory.getReceiverType(tree);
```

Comment:
`atmOfReceiverOrIdentifier` seems to always hold the type of a receiver, so I'm not sure why is has orIdentifier in its name.